zerver/lib/create_user.py

```diff
@@ -36,6 +36,13 @@ def get_display_email_address(user_profile: UserProfile, realm: Realm) -> str:
         return "user%s@%s" % (user_profile.id, get_fake_email_domain())
     return user_profile.delivery_email
 
+def get_role_for_new_user(invited_as: int, realm_creation: bool=False) -> int:
+    if invited_as == PreregistrationUser.INVITE_AS['REALM_ADMIN'] or realm_creation:
+        return UserProfile.ROLE_REALM_ADMINISTRATOR
+    elif invited_as == PreregistrationUser.INVITE_AS['GUEST_USER']:
+        return UserProfile.ROLE_GUEST
+    return UserProfile.ROLE_MEMBER
+
```

Comment:
I extracted this as a function as it was being used in both `zerver/views/registration.py` and `zproject/backends.py` and also helps in readability of code. 
I was not sure the exactly correct file to add this function, so added here as this seemed appropriate to me, please tell if we should change this to some other file.

Reply:
Yeah, this is a great extraction, and a fine place to put this.